### PR TITLE
Fix culture sensitive tests

### LINKED ISSUE #13212: Culture sensitive tests
Some tests are culture sensitive and fail on non en-US machines. It makes running test suites localy difficult when working on new PR.

Provide the steps required to reproduce the problem:

1. Run tests on a machine that is not installed with en-US culture
2. See it fail on UI message comparisons
image
(yes I'm french)

**Expected behavior**

Tests pass on all machines independently of their culture.

**Actual behavior**

Tests fail.

**Known workarounds**

Change the culture of the machine (tedious). Not run test suites (dangerous). 

**Related information**

Some tests fail due to warning/error message comparison, other due to number formatting ( "5,0" <> "5.0" )

### PR DESCRIPTION
**Pull request by thinkbeforecoding**
Fix culture sensitive tests. Fix #13212 

Some tests run FCS in process, in this case it uses the SetUICulture and SetCulture attributes.

For tests that run Fsc or Fsi as an external exe, it adds the --preferreduilang to force the culture.

Some tests still failed because the SR module caches the CurrentUICulture. In Fsi, an early call to SR.GetString is done before changing the culture indicated by the preferreduilang flag, so the local machine's default culture is cached for some error messages instead of using indicated culture. This as been fix by using CurrentUICulture directly.

### CODE REVIEW

File: tests/FSharp.Test.Utilities/Compiler.fs [hunk 1]
```diff
@@ -464,6 +464,9 @@ module rec Compiler =
         | FS fs -> FS { fs with IgnoreWarnings = true }
         | _ -> failwith "TODO: Implement ignorewarnings for the rest."
 
+    let withCulture culture (cUnit: CompilationUnit) : CompilationUnit =
```
Comment (vzarytovskii) [inline tests/FSharp.Test.Utilities/Compiler.fs:467]: It's a good helper, we can also, probably add `--preferreduilang:en-US` to the `defaultOptions`, so we, by default run in the `en-US` culture, and if we need a different one, we just use `withCulture`.

**Review by KevinRansom — APPROVED** [review]

**Review by vzarytovskii — APPROVED** [review]
Thanks for the fix @thinkbeforecoding, are the tests passing on your locale now, shall we merge it, or do you have any other changes pending?

**Comment by thinkbeforecoding** [comment]
You can merge it.
There are still 2 test failing locally but this is not related to culture. If I find why I will open a different PR.

**Comment by thinkbeforecoding** [comment]
🎉

**Pull request merged**